reverse_engineering/databaseService/databaseService.js

```diff
@@ -304,22 +304,42 @@ const getViewsIndexes = async (connectionClient, dbName) => {
 	);
 };
 
-const getPartitions = async (connectionClient, dbName, logger) => {
+const getTablesWhereClause = ({ schemaToTablesMap, schemaAlias, tableAlias }) => {
+	const schemas = Object.keys(schemaToTablesMap);
+
+	const tablesSelectionClauses = schemas.flatMap(schemaName =>
+		schemaToTablesMap[schemaName].map(
+			tableName => `(${schemaAlias}.name = '${schemaName}' AND ${tableAlias}.name = '${tableName}')`,
+		),
+	);
+	return `WHERE ${tablesSelectionClauses.join(' OR ')}`;
+};
+
+const getPartitions = async ({ connectionClient, tablesInfo, dbName, logger }) => {
 	const currentDbConnectionClient = await getNewConnectionClientByDb(connectionClient, dbName);
 
 	logger.log('info', { message: `Get '${dbName}' database partitions.` }, 'Reverse Engineering');
 
+	const tablesFilteringWhereClause = getTablesWhereClause({
+		schemaToTablesMap: tablesInfo,
+		schemaAlias: 'sch',
+		tableAlias: 'tbl',
+	});
+
 	return mapResponse(
 		await currentDbConnectionClient.query`
-		SELECT 
-			sch.name AS schemaName,
-			tbl.name AS tableName,
+		WITH user_selected_tables AS (
+        	SELECT tbl.object_id,tbl.name as tableName,sch.name as schemaName FROM sys.tables tbl JOIN sys.schemas sch ON sch.schema_id = tbl.schema_id
+        	${tablesFilteringWhereClause}
```

Comment:
I would make more sense to encapsulate the whole query instead of just the `WHERE` clause. That way, there is no need to pass those aliases as arguments.